relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,230 @@
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    #[serde(default)]
+    global: Option<GlobalConfig>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    pub global: bool,
+    // Upstream expects a list of public keys.
+    public_keys: Vec<()>,
+}
+
+impl GetGlobalConfig {
+    pub fn query() -> GetGlobalConfig {
+        GetGlobalConfig {
+            global: true,
+            public_keys: vec![],
+        }
+    }
+}
+
+impl UpstreamQuery for GetGlobalConfig {
+    type Response = GetGlobalConfigResponse;
+
+    fn method(&self) -> reqwest::Method {
+        Method::POST
+    }
+
+    fn path(&self) -> std::borrow::Cow<'static, str> {
+        Cow::Borrowed("/api/0/relays/projectconfigs/?version=4")
+    }
+
+    fn retry() -> bool {
+        false
+    }
+
+    fn priority() -> super::upstream::RequestPriority {
+        RequestPriority::High
+    }
+
+    fn route(&self) -> &'static str {
+        "global_config"
+    }
+}
+
+/// The message for requesting the most recent global config from [`GlobalConfigService`].
+pub struct Get;
+
+/// The message for receiving a watch that subscribes to the [`GlobalConfigService`].
+pub struct Subscribe;
+
+/// A way to get updates of the global config.
+pub enum GlobalConfigManager {
+    /// Returns the most recent global config.
+    Get(Sender<Arc<GlobalConfig>>),
+    /// Returns a [`watch::Receiver`] where global config updates will be sent to.
+    Subscribe(Sender<watch::Receiver<Arc<GlobalConfig>>>),
+}
+
+impl Interface for GlobalConfigManager {}
+
+impl FromMessage<Get> for GlobalConfigManager {
+    type Response = AsyncResponse<Arc<GlobalConfig>>;
+
+    fn from_message(_: Get, sender: Sender<Arc<GlobalConfig>>) -> Self {
+        Self::Get(sender)
+    }
+}
+
+impl FromMessage<Subscribe> for GlobalConfigManager {
+    type Response = AsyncResponse<watch::Receiver<Arc<GlobalConfig>>>;
+
+    fn from_message(_: Subscribe, sender: Sender<watch::Receiver<Arc<GlobalConfig>>>) -> Self {
+        Self::Subscribe(sender)
+    }
+}
+
+/// Service implementing the [`GlobalConfigManager`] interface.
+///
+/// The service offers two alternatives to fetch the [`GlobalConfig`]:
+/// responding to a [`Get`] message with the config for one-off requests, or
+/// subscribing to updates with [`Subscribe`] to keep up-to-date.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    config: Arc<Config>,
+    /// Sender of the [`watch`] channel for the subscribers of the service.
+    sender: watch::Sender<Arc<GlobalConfig>>,
+    /// Sender of the internal channel to forward global configs from upstream.
+    internal_tx: UnboundedSender<UpstreamResult>,
+    /// Receiver of the internal channel to forward global configs from upstream.
+    internal_rx: UnboundedReceiver<UpstreamResult>,
```

Comment:
wouldn't this struct be a bit cleaner if we created the channel in `spawn_handler` and cloned the sender in the function call?